src/Cake.Core/IO/FilePathCollection.cs

```diff
@@ -27,8 +27,8 @@ public sealed class FilePathCollection : IEnumerable<FilePath>
         /// <summary>
         /// Initializes a new instance of the <see cref="FilePathCollection"/> class.
         /// </summary>
-        /// <param name="comparer">The comparer.</param>
-        public FilePathCollection(PathComparer comparer)
+        /// <param name="comparer">The comparer. Defaults to <see cref="P:PathComparer.Default" />.</param>
+        public FilePathCollection(PathComparer comparer = null)
```

Comment:
@cake-build/cake-team please review potential breaking change.

Reply:
See my response below.

Reply:
This will retain API compatibility, but will break  ABI which means any addin or module utilizing this will throw an runtime error. Only non breaking way to add it would be to add a new parameterless contructor that forwards null to the existing construction.